test/Microsoft.Extensions.DependencyInjection.Tests/ServiceProviderValidationTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Tests
+{
+    public class ServiceProviderValidationTests
+    {
+        [Fact]
+        public void GetService_Throws_WhenScopedIsInjectedIntoSingleton()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
```

Comment:
Could we get tests with a couple more variants

* IFoo (Singleton) -> IBar (Singleton) ->  IBaz (Scoped)
* IFoo (Singleton) -> IBar (Transient) ->  IBaz (Scoped)